src/xenserver/opt/xensource/www/asset/deploy.js

```diff
@@ -0,0 +1,205 @@
+let call,
+  vmRef,
+  vmIp,
+  host = window.location.host
+
+function _call(method, params) {
+  console.log(`-> ${method}(${JSON.stringify(params)})`)
+  return window
+    .fetch(`//${host}/jsonrpc`, {
+      body: JSON.stringify({
+        id: 0,
+        jsonrpc: '2.0',
+        method,
+        params
+      }),
+      method: 'POST',
+      headers: new Headers({
+        Accept: 'application/json',
+        'Content-Type': 'application/json'
+      })
+    })
+    .then(res => {
+      if (!res.ok) {
+        throw res.status
+      }
+      return res.json().then(json => {
+        if (json.result !== undefined) {
+          console.log(`<- ${JSON.stringify(json.result)}`)
+          return json.result
+        }
+        console.error(`<x ${JSON.stringify(json.error)}`)
+        throw json.error
+      })
+    })
+    .catch(error => {
+      if (error && error.message === 'HOST_IS_SLAVE') {
+        console.log(
+          'Host is slave, changing host to master',
+          error.data[0]
+        )
+        host = error.data[0]
+        return _call(method, params)
+      }
+      throw error
+    })
+}
+
+function connect() {
+  $('#connect fieldset').attr('disabled', true)
+  _call('session.login_with_password', [
+    'root',
+    $('#pwd').val(),
+    '2.3',
+    'XOA deploy'
+  ])
+    .then(sessionRef => {
+      call = (method, ...params) =>
+        _call(method, [sessionRef].concat(params))
+    })
+    .then(() => call('SR.get_all_records'))
+    .then(srs => {
+      $('#connect').css({ display: 'none' })
+      $('#config').css({ display: 'block' })
+      const select = $('#srs')
+      let sr
+      Object.keys(srs).forEach(srRef => {
+        sr = srs[srRef]
+        if (sr.content_type !== 'iso' && sr.physical_size > 0) {
+          select.append(
+            $('<option/>')
+              .val(srRef)
+              .text(
+                `${sr.name_label} - ${Math.round(
+                  (sr.physical_size - sr.physical_utilisation) /
+                    Math.pow(1024, 3)
+                )} GiB left`
+              )
+          )
+        }
+      })
+    })
+    .catch(error => {
+      alert(JSON.stringify(error, null, 2))
+    })
+}
+
+function deploy() {
+  const status = text => $('#deploy').text(text)
+  const srRef = $('#srs').val()
+  status('Deploying XOA...')
+  $('#config fieldset').attr('disabled', true)
+  call(
+    'VM.import',
+    'http://xoa.io:8888/',
```

Comment:
I think HTTPS would work for web deploy (because it's at last a recent XAPI accepting HTTPS). @julien-f should we do anything to have HTTPS available for download on our side?

Reply:
Just use `https://xoa.io/xoa.xva`.